packages/e2e/lib/e2e_driver.dart

```diff
@@ -4,15 +4,49 @@ import 'dart:io';
 import 'package:e2e/common.dart' as e2e;
 import 'package:flutter_driver/flutter_driver.dart';
 
-Future<void> main() async {
+Future<void> main() async => e2eDriver();
+
+/// Adaptor to run E2E test using `flutter drive`.
+///
+/// For an E2E test `<test_name>.dart`, put a file named `<test_name>_test.dart`
+/// in the app's `test_driver` directory:
+///
+/// ```dart
+/// import 'dart:async';
+///
+/// import 'package:e2e/e2e_driver.dart' as e2e;
+///
+/// Future<void> main() async => e2e.e2eDriver();
+///
+/// ```
+Future<void> e2eDriver({
+  Duration timeout = const Duration(minutes: 1),
+  String testName,
+  bool traceTimeline = false,
+}) async {
+  print(testName);
   final FlutterDriver driver = await FlutterDriver.connect();
-  final String jsonResult =
-      await driver.requestData(null, timeout: const Duration(minutes: 1));
+  String jsonResult;
+  Timeline timeline;
+  Future<void> runner() async {
+    jsonResult = await driver.requestData(null, timeout: timeout);
+  }
+
+  if (traceTimeline) {
+    timeline = await driver.traceAction(runner);
+  } else {
+    await runner();
+  }
   final e2e.Response response = e2e.Response.fromJson(jsonResult);
```

Comment:
Won't this now be null sometimes and throw?

Reply:
Why? `runner` is awaited either directly or in `traceAction`.

Reply:
Ahh I see now. 

This is a nit, but IMO it would read a bit more clearly if you just directly used start/stop trace if needed, rather than passing a closure to a convenience wrapper.

Reply:
`FlutterDriver` is an abstract class. We have two implementation of it and all override `traceAction`. I would rather still use this method, but add some comments here.